### PR TITLE
Update bullseye version of builder to v1.23.1

### PR DESCRIPTION
**Pull request by mike-kelly-ch**
As per title.

### CODE REVIEW

File: builder/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.19.6-bullseye as builder
+FROM golang:1.23.1-bullseye as builder
```
Comment (JoelClemence) [inline builder/Dockerfile]: As per the PR on @companieshouse/ci-golang-build#16 - Is there a reason why we're pinning to the minor version too? 
```suggestion
FROM golang:1.23-bullseye as builder
```

Comment (mike-kelly-ch) [reply builder/Dockerfile]: 👍

**Review by bcullerton — CHANGES_REQUESTED** [review]
If we are sticking with this build then the version file needs updating in builder/version.

I would strongly prefer to use the same builder image for ci and local builds however

**Comment by mike-kelly-ch** [comment]
> If we are sticking with this build then the version file needs updating in builder/version.
> 
> I would strongly prefer to use the same builder image for ci and local builds however

Investigating.

**Review by JoelClemence — APPROVED** [review]

**Pull request closed without merging**